test/hotspot/jtreg/compiler/stringopts/TestFetchStaticField.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8267773
+ * @summary test that fetching a static field produces the correct result
+ * @library /test/lib /
+ * @run main compiler.stringopts.TestFetchStaticField
+ */
+
+package compiler.stringopts;
+
+import java.lang.StringIndexOutOfBoundsException;
+import jdk.test.lib.Asserts;
+
+public class TestFetchStaticField {
```

Comment:
Why `TestFetchStaticField`? Is this really specific to static fields?

Reply:
No - it's a leftover from when I was on the completely wrong track :P

I removed this test and added yours instead.